src/ServiceProvider.php

```diff
@@ -160,15 +161,31 @@ private function registerCollections()
         $this->app->bind('statamic.eloquent.collections.model', function () {
             return config('statamic.eloquent-driver.collections.model');
         });
+    }
+
+    private function registerCollectionTrees()
+    {
+        // Use the collections' driver as a fallback to ensure brackwards compatibility.
+        $collectionsDriver = config('statamic.eloquent-driver.collections.driver', 'file');
+
+        if (config('statamic.eloquent-driver.collection_trees.driver', $collectionsDriver) != 'eloquent') {
+            return;
+        }
 
         Statamic::repository(CollectionTreeRepositoryContract::class, CollectionTreeRepository::class);
 
         $this->app->bind('statamic.eloquent.collections.tree', function () {
-            return config('statamic.eloquent-driver.collections.tree');
+            // Use the old related config as a fallback to ensure backwards compatibility.
+            $collectionsOldTree = config('statamic.eloquent-driver.collections.tree');
+
+            return config('statamic.eloquent-driver.collection_trees.tree', $collectionsOldTree);
```

Comment:
You could make this a one-liner as follows:

`return config('statamic.eloquent-driver.collection_trees.tree', config('statamic.eloquent-driver.collections.tree'));`